File: CRM-Client/src/api/customerAICreate.ts
/**
 * AI 创建客户 API
 */
import request from '@/utils/request'

/* eslint-disable crmwolf/require-zod-schema */

export interface CustomerAICreateParseRequest {
  content: string
}

export interface CustomerAIParsedInfo {
  account_name: string | null
  city: string | null
  company_scale: string | null
  source: string | null
  industry_hint: string | null
  missing_fields: string[]
}

export interface CustomerAIContactInfo {
  contact_name: string | null
  contact_phone: string | null
  contact_position: string | null
  contact_gender: string | null
  contact_email: string | null
}

export interface CustomerAIFollowUpInfo {
  content: string | null
  next_action: string | null
  next_follow_time: string | null
}

export interface CustomerAIParseSSEEvent {
  event: 'status' | 'content' | 'parsed' | 'error'
  message?: string
  content?: string
  customer_info?: CustomerAIParsedInfo
  contact_info?: CustomerAIContactInfo
  follow_up_info?: CustomerAIFollowUpInfo
  thinking_process?: string
}

export interface CustomerAICreateRequest {
  customer_info: CustomerAIParsedInfo
  contact_info: CustomerAIContactInfo
  follow_up_info?: CustomerAIFollowUpInfo
}

export const customerAiCreateApi = {
  /**
   * AI 解析客户创建信息（SSE 流式响应）
   */
  parseSSE: async (
    data: CustomerAICreateParseRequest,
    onEvent: (event: CustomerAIParseSSEEvent) => void,
    token: string
  ): Promise<void> => {
    const url = '/api/v1/customers/ai/create/parse'

    const response = await fetch(url, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Authorization': `Bearer ${token}`
      },
      body: JSON.stringify(data)
    })

    if (!response.ok) {
      throw new Error(`HTTP error: ${response.status}`)
    }

    const reader = response.body?.getReader()
    if (!reader) {
      throw new Error('No response body')
    }

    const decoder = new TextDecoder()
    let buffer = ''

    while (true) {
      const { done, value } = await reader.read()
      if (done) break

      buffer += decoder.decode(value, { stream: true })
      const lines = buffer.split('\n\n')
      buffer = lines.pop() ?? ''

      for (const line of lines) {
        if (line.startsWith('data: ')) {
          try {
            const eventData = JSON.parse(line.slice(6)) as CustomerAIParseSSEEvent
            onEvent(eventData)

            if (eventData.event === 'parsed' || eventData.event === 'error') {
              return
            }
          } catch {
            // 忽略解析错误
          }
        }
      }
    }
  },

  /**
   * 从 AI 解析结果创建客户
   */
  createFromAI: (data: CustomerAICreateRequest): Promise<{ id: number; account_name: string }> => {
    return request.post<{ id: number; account_name: string }>(
      '/v1/customers/ai/create/submit',
      data
    )
  }
}

File: CRM-Server/app/schemas/customer_ai_create.py
"""
AI 创建客户 Schema

用于 AI 智能创建客户功能
"""
from pydantic import BaseModel, Field
from typing import Optional


class CustomerAICreateParseRequest(BaseModel):
    """AI 解析客户创建信息请求"""
    content: str = Field(..., min_length=1, description="用户输入的自然语言描述")


class CustomerAIParsedInfo(BaseModel):
    """AI 解析出的客户信息"""
    account_name: Optional[str] = Field(None, description="客户公司名称")
    city: Optional[str] = Field(None, description="所在城市")
    company_scale: Optional[str] = Field(None, description="公司规模")
    source: Optional[str] = Field(None, description="客户来源")
    industry_hint: Optional[str] = Field(None, description="行业关键词")
    missing_fields: list[str] = Field(default_factory=list, description="缺失的必填字段")


class CustomerAIContactInfo(BaseModel):
    """AI 解析出的主联系人信息"""
    contact_name: Optional[str] = Field(None, description="联系人姓名")
    contact_phone: Optional[str] = Field(None, description="联系电话")
    contact_position: Optional[str] = Field(None, description="职务")
    contact_gender: Optional[str] = Field(None, description="性别：1=男, 2=女")
    contact_email: Optional[str] = Field(None, description="邮箱")


class CustomerAIFollowUpInfo(BaseModel):
    """AI 解析出的跟进记录信息"""
    content: Optional[str] = Field(None, description="跟进内容")
    next_action: Optional[str] = Field(None, description="下一步动作")
    next_follow_time: Optional[str] = Field(None, description="下次跟进时间")


class CustomerAICreateRequest(BaseModel):
    """AI 创建客户请求（用户确认后提交）"""
    customer_info: CustomerAIParsedInfo = Field(..., description="客户信息")
    contact_info: CustomerAIContactInfo = Field(..., description="主联系人信息")
    follow_up_info: Optional[CustomerAIFollowUpInfo] = Field(None, description="跟进信息（可选）")
